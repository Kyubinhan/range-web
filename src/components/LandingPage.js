import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { logout } from '../actions/authActions';
import { getReferences } from '../actions/commonActions';
import { getZones } from '../actions/rangeUsePlanActions';
import Navbar from './Navbar';

const propTypes = {
  component: PropTypes.func.isRequired,
};

export class LandingPage extends Component {
  componentWillMount() {
    const { getReferences, getZones } = this.props;
    getReferences();
    getZones();
  }
  
  onLogout = () => {
    this.props.logout();
  }

  render() {
    const { component: Component, user, ...rest } = this.props;

    return (
      <div className="main">
        <Navbar onLogout={this.onLogout}/>
        <Component {...rest} />
        <footer></footer>
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { user } = state.auth;
  
  return {
    user
  };  
};

LandingPage.propTypes = propTypes;
export default connect (
  mapStateToProps, { logout, getReferences, getZones }
)(LandingPage)