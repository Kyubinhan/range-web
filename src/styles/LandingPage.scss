@import 'styles/mixins.scss';
@import 'styles/Grid.scss';

.navbar {
  background: $primary-color;
  border-bottom: 2px solid $accent-color;

  &__container {
    @extend .container;
    height: $nav-height;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  &__left,
  &__right {
    display: flex;
    align-items: center;
  }

  &__link {
    height: $nav-height;
    display: flex;
    align-items: center;
    color: white;
    padding: 0 20px;
    cursor: pointer;
    
    &:hover {
      color: white;
      text-decoration: underline;
    }

    &--active {
      background: $primary-light-color;
    }
  }

  &__logo {
    height: 40px;
  }

  &__title {
    border-left: 1px solid $accent-color;
    font-size: 1.2rem;
    padding-left: 15px;
    margin-left: 10px;
    color: white;
    white-space: nowrap;
  }
}

.toasts {
  top: 30px;
  position: fixed;
  right: 30px;
  width: 400px;
  z-index: $toast-z-index;
}

.toast {
  min-height: 65px;
  border-radius: 5px;
  color: black;
  display: flex;
  align-items: stretch;
  border: 1px solid rgba(34,36,38,.1);

  &:not(:last-child) {
    margin: 0 0 12px;
  }

  &__icon {
    min-width: 50px;
    max-width: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
    color:white;
    &__success {
      background-color: green;
    }
  
    &__error {
      background-color: $error-color;
    }
  }

  &__content {
    display: flex;
    justify-content: center;
    align-items: center;
    flex: 1;
    background-color: white;
    overflow: hidden;
    text-overflow: ellipsis;
    font-size: 1.1rem;
    padding: 15px;
  }

  &__dismiss {
    -webkit-appearance: none;
    -moz-appearance: none;
    background: white;
    border: 0;
    color: inherit;
    cursor: pointer;
    display: block;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
    font: inherit;
    padding: 10px;
    font-size: 1.3rem;
  }
}

.offline {
  background-color: $error-color;
  color: white;
  text-align: center;
  height: 15px;
  font-size: 0.9rem;
  display: flex;
  align-items: center;
  justify-content: center;
}

.user-error {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
  font-size: 1.5rem;
}

footer {
  height: 20px;
}

.confirmation-modal {
  &__content {
    font-size: 1.2rem;
  }
}