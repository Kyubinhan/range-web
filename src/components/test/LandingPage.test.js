import React from 'react';
import { shallow } from 'enzyme';
import { LandingPage } from '../LandingPage';

const props = {};
const setupProps = () => {
  props.location = {};
  props.match = {};
  props.history = {};
  props.component = jest.fn();
  props.logout = jest.fn();
  props.getReferences = jest.fn();
  props.getZones = jest.fn();
};

beforeEach(() => {
  setupProps();
});

describe('LandingPage', () => {
  xit('renders correctly', () => {
    const wrapper = shallow(<LandingPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Event handlers', () => {
    it('`onLogout` calls props.logout', () => {
      const wrapper = shallow(<LandingPage {...props} />);
      
      wrapper.instance().onLogout();
      expect(props.logout).toHaveBeenCalled();
    });
  });

  describe('Life cycles', () => {
    it('componentDidMount calls the right functinos', () => {
      const wrapper = shallow(<LandingPage {...props} />);
      
      expect(props.getReferences).toHaveBeenCalled();
      expect(props.getZones).toHaveBeenCalled();
    });
  });
});