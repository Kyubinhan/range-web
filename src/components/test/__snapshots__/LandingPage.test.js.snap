// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage
    component={[Function]}
    history={Object {}}
    location={Object {}}
    logout={[Function]}
    match={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="nav"
>
          <div
                    className="nav__left"
          >
                    <div
                              className="nav__icon"
                    >
                              <Icon
                                        as="i"
                                        name="bars"
                                        onClick={[Function]}
                                        size="large"
                              />
                    </div>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav__title"
                              to="/home"
                    >
                              MyRA
                    </NavLink>
          </div>
          <div
                    className="nav__right"
          >
                    <div
                              className="avatar"
                    >
                              <div
                                        className="avatar__initial"
                              >
                                        KH
                              </div>
                    </div>
          </div>
</div>,
        <div
          className="main"
>
          <nav
                    className="sidebar sidebar--hidden"
          >
                    <div
                              className="sidebar__header"
                    />
                    <div
                              className="sidebar__list"
                    >
                              <NavLink
                                        activeClassName="sidebar__list__item--active"
                                        ariaCurrent="true"
                                        className="sidebar__list__item"
                                        to="/home"
                              >
                                        Home
                              </NavLink>
                              <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        className="sidebar__list__item"
                                        id="sign-out"
                                        onClick={[Function]}
                                        to="/login"
                              >
                                        Sign Out
                              </NavLink>
                    </div>
          </nav>
          <div
                    className="overlay overlay--hidden"
                    onClick={[Function]}
          />
          <div
                    className="content"
          >
                    <mockConstructor
                              history={Object {}}
                              location={Object {}}
                              logout={[Function]}
                              match={Object {}}
                    />
          </div>
</div>,
      ],
      "className": "landing-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="nav__left"
>
              <div
                            className="nav__icon"
              >
                            <Icon
                                          as="i"
                                          name="bars"
                                          onClick={[Function]}
                                          size="large"
                            />
              </div>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="nav__title"
                            to="/home"
              >
                            MyRA
              </NavLink>
</div>,
            <div
              className="nav__right"
>
              <div
                            className="avatar"
              >
                            <div
                                          className="avatar__initial"
                            >
                                          KH
                            </div>
              </div>
</div>,
          ],
          "className": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="nav__icon"
>
                  <Icon
                                    as="i"
                                    name="bars"
                                    onClick={[Function]}
                                    size="large"
                  />
</div>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav__title"
                  to="/home"
>
                  MyRA
</NavLink>,
              ],
              "className": "nav__left",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Icon
                    as="i"
                    name="bars"
                    onClick={[Function]}
                    size="large"
/>,
                  "className": "nav__icon",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "i",
                    "name": "bars",
                    "onClick": [Function],
                    "size": "large",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "MyRA",
                  "className": "nav__title",
                  "to": "/home",
                },
                "ref": null,
                "rendered": "MyRA",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="avatar"
>
                <div
                                className="avatar__initial"
                >
                                KH
                </div>
</div>,
              "className": "nav__right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="avatar__initial"
>
                  KH
</div>,
                "className": "avatar",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "KH",
                  "className": "avatar__initial",
                },
                "ref": null,
                "rendered": "KH",
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <nav
              className="sidebar sidebar--hidden"
>
              <div
                            className="sidebar__header"
              />
              <div
                            className="sidebar__list"
              >
                            <NavLink
                                          activeClassName="sidebar__list__item--active"
                                          ariaCurrent="true"
                                          className="sidebar__list__item"
                                          to="/home"
                            >
                                          Home
                            </NavLink>
                            <NavLink
                                          activeClassName="active"
                                          ariaCurrent="true"
                                          className="sidebar__list__item"
                                          id="sign-out"
                                          onClick={[Function]}
                                          to="/login"
                            >
                                          Sign Out
                            </NavLink>
              </div>
</nav>,
            <div
              className="overlay overlay--hidden"
              onClick={[Function]}
/>,
            <div
              className="content"
>
              <mockConstructor
                            history={Object {}}
                            location={Object {}}
                            logout={[Function]}
                            match={Object {}}
              />
</div>,
          ],
          "className": "main",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="sidebar__header"
/>,
                <div
                  className="sidebar__list"
>
                  <NavLink
                                    activeClassName="sidebar__list__item--active"
                                    ariaCurrent="true"
                                    className="sidebar__list__item"
                                    to="/home"
                  >
                                    Home
                  </NavLink>
                  <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="sidebar__list__item"
                                    id="sign-out"
                                    onClick={[Function]}
                                    to="/login"
                  >
                                    Sign Out
                  </NavLink>
</div>,
              ],
              "className": "sidebar sidebar--hidden",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "sidebar__header",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <NavLink
                      activeClassName="sidebar__list__item--active"
                      ariaCurrent="true"
                      className="sidebar__list__item"
                      to="/home"
>
                      Home
</NavLink>,
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="sidebar__list__item"
                      id="sign-out"
                      onClick={[Function]}
                      to="/login"
>
                      Sign Out
</NavLink>,
                  ],
                  "className": "sidebar__list",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "sidebar__list__item--active",
                      "ariaCurrent": "true",
                      "children": "Home",
                      "className": "sidebar__list__item",
                      "to": "/home",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "ariaCurrent": "true",
                      "children": "Sign Out",
                      "className": "sidebar__list__item",
                      "id": "sign-out",
                      "onClick": [Function],
                      "to": "/login",
                    },
                    "ref": null,
                    "rendered": "Sign Out",
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "nav",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "overlay overlay--hidden",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <mockConstructor
                history={Object {}}
                location={Object {}}
                logout={[Function]}
                match={Object {}}
/>,
              "className": "content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "history": Object {},
                "location": Object {},
                "logout": [Function],
                "match": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="nav"
>
            <div
                        className="nav__left"
            >
                        <div
                                    className="nav__icon"
                        >
                                    <Icon
                                                as="i"
                                                name="bars"
                                                onClick={[Function]}
                                                size="large"
                                    />
                        </div>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="nav__title"
                                    to="/home"
                        >
                                    MyRA
                        </NavLink>
            </div>
            <div
                        className="nav__right"
            >
                        <div
                                    className="avatar"
                        >
                                    <div
                                                className="avatar__initial"
                                    >
                                                KH
                                    </div>
                        </div>
            </div>
</div>,
          <div
            className="main"
>
            <nav
                        className="sidebar sidebar--hidden"
            >
                        <div
                                    className="sidebar__header"
                        />
                        <div
                                    className="sidebar__list"
                        >
                                    <NavLink
                                                activeClassName="sidebar__list__item--active"
                                                ariaCurrent="true"
                                                className="sidebar__list__item"
                                                to="/home"
                                    >
                                                Home
                                    </NavLink>
                                    <NavLink
                                                activeClassName="active"
                                                ariaCurrent="true"
                                                className="sidebar__list__item"
                                                id="sign-out"
                                                onClick={[Function]}
                                                to="/login"
                                    >
                                                Sign Out
                                    </NavLink>
                        </div>
            </nav>
            <div
                        className="overlay overlay--hidden"
                        onClick={[Function]}
            />
            <div
                        className="content"
            >
                        <mockConstructor
                                    history={Object {}}
                                    location={Object {}}
                                    logout={[Function]}
                                    match={Object {}}
                        />
            </div>
</div>,
        ],
        "className": "landing-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="nav__left"
>
                <div
                                className="nav__icon"
                >
                                <Icon
                                                as="i"
                                                name="bars"
                                                onClick={[Function]}
                                                size="large"
                                />
                </div>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="nav__title"
                                to="/home"
                >
                                MyRA
                </NavLink>
</div>,
              <div
                className="nav__right"
>
                <div
                                className="avatar"
                >
                                <div
                                                className="avatar__initial"
                                >
                                                KH
                                </div>
                </div>
</div>,
            ],
            "className": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="nav__icon"
>
                    <Icon
                                        as="i"
                                        name="bars"
                                        onClick={[Function]}
                                        size="large"
                    />
</div>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav__title"
                    to="/home"
>
                    MyRA
</NavLink>,
                ],
                "className": "nav__left",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Icon
                      as="i"
                      name="bars"
                      onClick={[Function]}
                      size="large"
/>,
                    "className": "nav__icon",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "i",
                      "name": "bars",
                      "onClick": [Function],
                      "size": "large",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "MyRA",
                    "className": "nav__title",
                    "to": "/home",
                  },
                  "ref": null,
                  "rendered": "MyRA",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="avatar"
>
                  <div
                                    className="avatar__initial"
                  >
                                    KH
                  </div>
</div>,
                "className": "nav__right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="avatar__initial"
>
                    KH
</div>,
                  "className": "avatar",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "KH",
                    "className": "avatar__initial",
                  },
                  "ref": null,
                  "rendered": "KH",
                  "type": "div",
                },
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <nav
                className="sidebar sidebar--hidden"
>
                <div
                                className="sidebar__header"
                />
                <div
                                className="sidebar__list"
                >
                                <NavLink
                                                activeClassName="sidebar__list__item--active"
                                                ariaCurrent="true"
                                                className="sidebar__list__item"
                                                to="/home"
                                >
                                                Home
                                </NavLink>
                                <NavLink
                                                activeClassName="active"
                                                ariaCurrent="true"
                                                className="sidebar__list__item"
                                                id="sign-out"
                                                onClick={[Function]}
                                                to="/login"
                                >
                                                Sign Out
                                </NavLink>
                </div>
</nav>,
              <div
                className="overlay overlay--hidden"
                onClick={[Function]}
/>,
              <div
                className="content"
>
                <mockConstructor
                                history={Object {}}
                                location={Object {}}
                                logout={[Function]}
                                match={Object {}}
                />
</div>,
            ],
            "className": "main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="sidebar__header"
/>,
                  <div
                    className="sidebar__list"
>
                    <NavLink
                                        activeClassName="sidebar__list__item--active"
                                        ariaCurrent="true"
                                        className="sidebar__list__item"
                                        to="/home"
                    >
                                        Home
                    </NavLink>
                    <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        className="sidebar__list__item"
                                        id="sign-out"
                                        onClick={[Function]}
                                        to="/login"
                    >
                                        Sign Out
                    </NavLink>
</div>,
                ],
                "className": "sidebar sidebar--hidden",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "sidebar__header",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <NavLink
                        activeClassName="sidebar__list__item--active"
                        ariaCurrent="true"
                        className="sidebar__list__item"
                        to="/home"
>
                        Home
</NavLink>,
                      <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="sidebar__list__item"
                        id="sign-out"
                        onClick={[Function]}
                        to="/login"
>
                        Sign Out
</NavLink>,
                    ],
                    "className": "sidebar__list",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "activeClassName": "sidebar__list__item--active",
                        "ariaCurrent": "true",
                        "children": "Home",
                        "className": "sidebar__list__item",
                        "to": "/home",
                      },
                      "ref": null,
                      "rendered": "Home",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "activeClassName": "active",
                        "ariaCurrent": "true",
                        "children": "Sign Out",
                        "className": "sidebar__list__item",
                        "id": "sign-out",
                        "onClick": [Function],
                        "to": "/login",
                      },
                      "ref": null,
                      "rendered": "Sign Out",
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "nav",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "overlay overlay--hidden",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <mockConstructor
                  history={Object {}}
                  location={Object {}}
                  logout={[Function]}
                  match={Object {}}
/>,
                "className": "content",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "history": Object {},
                  "location": Object {},
                  "logout": [Function],
                  "match": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
